GarbageCollectionApp/GarbageCollectionApp/Controllers/RouteController.cs

```diff
@@ -0,0 +1,118 @@
+            {
+                Latitude = 45.7877059,
+                Longitude = 24.0247875,
+                Address = "Groapa de gunoi Cristian, DN1 FN, Cristian 557085",
+                CollectionTime = date
+            };
+
+            var points = await db.GarbageCollections
+                .Where(g => g.CollectionTime.Date == date.Date)
+                .OrderBy(g => g.CollectionTime)
+                .Select(g => new Coordinate
+                {
+                    Latitude = g.Latitude,
+                    Longitude = g.Longitude,
+                    Address = g.Address,
+                    CollectionTime = g.CollectionTime
+                })
+                .ToListAsync();
+
+            var allPoints = new List<Coordinate> { startPoint }
+                .Concat(points)
+                .Append(finishPoint)
+                .ToList();
+            // uses LoadMatrixService to get the distance matrix
+            var distances = matrixService.GetDistanceMatrix();
+
+            // statistics will be returned as json
+            var optimizedOrder = OptimizeRoute(distances);
+            var originalOrder = Enumerable.Range(0, distances.Length).ToList();
+            
+            double originalDistance = RouteDistance(originalOrder, distances);
+            double optimizedDistance = RouteDistance(optimizedOrder, distances);
+
+            double originalTimeMin = EstimateRouteTimeMinutes(originalOrder, distances);
+            double optimizedTimeMin = EstimateRouteTimeMinutes(optimizedOrder, distances);
+
+            return Json(new
+            {
+                allPoints,
+                optimizedOrder,
+                originalDistance = originalDistance / 1000.0,
+                totalDistance = optimizedDistance / 1000.0,
+                originalTimeMinutes = Math.Round(originalTimeMin),
+                estimatedTimeMinutes = Math.Round(optimizedTimeMin)
+            });
+        }
+
+        private List<int> OptimizeRoute(double[][] distances)
+        {
+            var route = new List<int> { 0 };
+            var unvisited = Enumerable.Range(1, distances.Length - 1).ToList();
+
+            while (unvisited.Count > 0)
+            {
+                var last = route.Last();
+                var next = unvisited.OrderBy(i => distances[last][i]).First();
+                route.Add(next);
+                unvisited.Remove(next);
+            }
+
+            return route;
+        }
+        private double RouteDistance(List<int> route, double[][] dist)
+        {
+            double sum = 0;
+            for (int i = 0; i < route.Count - 1; i++)
+            {
+                sum += dist[route[i]][route[i + 1]];
+            }
+            return sum;
+        }
+        //method to estimate the time it takes to run the route, considering an average speed of 20 km/h
+        private double EstimateRouteTimeMinutes(List<int> routeOrder, double[][] distanceMatrix, double averageSpeedKmh = 20.0)
```

Comment:
Foarte tare ideea de a calcula timpul in functie de o viteza medie. Cum ati putea calcula pe bune timpul de parcurgere al traseului? (Fara sa presupuneti o viteza medie)